app/src/test/java/org/oppia/app/player/exploration/ExplorationActivityLocalTest.kt

```diff
@@ -0,0 +1,178 @@
+import org.oppia.domain.exploration.ExplorationDataController
+import org.oppia.domain.exploration.TEST_EXPLORATION_ID_30
+import org.oppia.domain.question.QuestionModule
+import org.oppia.domain.topic.TEST_STORY_ID_0
+import org.oppia.domain.topic.TEST_TOPIC_ID_0
+import org.oppia.testing.FakeEventLogger
+import org.oppia.testing.TestAccessibilityModule
+import org.oppia.testing.TestDispatcherModule
+import org.oppia.testing.TestLogReportingModule
+import org.oppia.util.caching.CacheAssetsLocally
+import org.oppia.util.gcsresource.GcsResourceModule
+import org.oppia.util.logging.LoggerModule
+import org.oppia.util.networking.NetworkConnectionUtil
+import org.oppia.util.parser.GlideImageLoaderModule
+import org.oppia.util.parser.HtmlParserEntityTypeModule
+import org.oppia.util.parser.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val TEST_TOPIC_ID = "GJ2rLXRKD5hw"
+private const val TEST_STORY_ID = "GJ2rLXRKD5hw"
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = ExplorationActivityLocalTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class ExplorationActivityLocalTest {
+
+  @Inject
+  lateinit var fakeEventLogger: FakeEventLogger
+
+  private lateinit var networkConnectionUtil: NetworkConnectionUtil
+  private lateinit var explorationDataController: ExplorationDataController
+  private val internalProfileId: Int = 0
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testExploration_onLaunch_logsEvent() {
+    getApplicationDependencies(TEST_EXPLORATION_ID_30)
+    launch<ExplorationActivity>(
+      createExplorationActivityIntent(
+        internalProfileId,
+        TEST_TOPIC_ID_0,
+        TEST_STORY_ID_0,
+        TEST_EXPLORATION_ID_30
+      )
+    ).use {
+
+      val event = fakeEventLogger.getMostRecentEvent()
+
+      assertThat(event.context.activityContextCase).isEqualTo(EXPLORATION_CONTEXT)
+      assertThat(event.actionName).isEqualTo(EventLog.EventAction.OPEN_EXPLORATION_ACTIVITY)
+      assertThat(event.priority).isEqualTo(EventLog.Priority.ESSENTIAL)
+      assertThat(event.context.explorationContext.explorationId).matches(TEST_EXPLORATION_ID_30)
+      assertThat(event.context.explorationContext.topicId).matches(TEST_TOPIC_ID_0)
+      assertThat(event.context.explorationContext.storyId).matches(TEST_STORY_ID_0)
+
+    }
+  }
+
+  private fun getApplicationDependencies(id: String) {
+    launch(ExplorationInjectionActivity::class.java).use {
+      it.onActivity { activity ->
+        networkConnectionUtil = activity.networkConnectionUtil
+        explorationDataController = activity.explorationDataController
+        explorationDataController.startPlayingExploration(id)
+      }
+    }
+  }
+
+  private fun launchExplorationActivity(internalProfileId: Int) {
+
```

Comment:
Unexpected blank line(s) before `}`. Also if this method is not in use, can we remove it?

Reply:
Done.